src/components/general/ButtonLayout.js

```diff
@@ -0,0 +1,142 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { View, Dimensions } from "react-native";
+
+const dimension = Dimensions.get("window");
+
+class ButtonLayout extends Component {
+	constructor(props) {
+		super(props);
+	}
+
+	static propTypes = {
+		icon: PropTypes.shape({})
+	}
+
+	render() {
+		const {
+			buttonContainer,
+			doubleLayer
+		} = styles;
+
+		const {
+			icon,
+			children,
+			containerStyle
+		} = this.props;
+
+		let i = 0;
+		let realButtons = [];
+		let length;
+		let layers = [];
+		let iconVar;
+
+		if (!children) {
+			return null;
+		}
+		else if (!children.length) {
+			if (children) {
+				layers.push(
+					this.renderSingleButton(children)
+				);
+			}
+			// else return null;
+		}
+		else {
+			children.forEach(function(child) {
+				if (child) realButtons.push(child);
+			});
+		}
+
+		length = realButtons.length;
+
+		if (length % 2 === 1) {
+			layers.push(
+				this.renderSingleButton(realButtons[i])
+			);
+			i++;
+		}
+
+		for (; i + 2 <= length;) {
+			iconVar = null;
+
+			if (icon && Math.trunc(i / 2) + 1 === icon.layer)
+				iconVar = icon.data;
+
+			layers.push(
+				<View style = { doubleLayer }>
+					<View style = { buttonContainer }>
+					    { realButtons[i] }
+					</View>
+					{ iconVar }
+					<View style = { buttonContainer }>
+					    { realButtons[i + 1] }
+					</View>
+				</View>
+			);
+			i = i + 2;
```

Comment:
I'm going to change this

Reply:
What are you changing?

Reply:
I'm going to refactor that page to look cleaner